overlord/snapstate/snapstate.go

```diff
@@ -2506,6 +2506,17 @@ func updateWithDeviceContext(st *state.State, name string, opts *RevisionOptions
 	switchCohortKey := snapst.CohortKey != opts.CohortKey
 	toggleIgnoreValidation := snapst.IgnoreValidation != flags.IgnoreValidation
 	if infoErr == store.ErrNoUpdateAvailable && (switchChannel || switchCohortKey || toggleIgnoreValidation) {
+		currentInfo, err := snapst.CurrentInfo()
+		if err != nil {
+			return nil, err
+		}
+
+		// if there isn't an update available, then we should sill add the
+		// current info to the prereq tracker. this is because we might not
```

Comment:
I think at this point inside the if we know  s/might not/will not/ no?

Reply:
There are a couple places that could return an error. I moved the call to `addPrereq` closer to the end of the function to eliminate those. The only other remaining place for an error would be because of programmer error, I believe.